runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SdkComponents.java

```diff
@@ -88,11 +88,13 @@ public static SdkComponents create(RunnerApi.Components components) {
   public static SdkComponents create(PipelineOptions options) {
     SdkComponents sdkComponents = new SdkComponents(RunnerApi.Components.getDefaultInstance(), "");
     PortablePipelineOptions portablePipelineOptions = options.as(PortablePipelineOptions.class);
-    sdkComponents.defaultEnvironmentId =
-        sdkComponents.registerEnvironment(
-            Environments.createOrGetDefaultEnvironment(
+    sdkComponents.registerEnvironment(
+        Environments.createOrGetDefaultEnvironment(
                 portablePipelineOptions.getDefaultEnvironmentType(),
-                portablePipelineOptions.getDefaultEnvironmentConfig()));
+                portablePipelineOptions.getDefaultEnvironmentConfig())
+            .toBuilder()
+            .addAllDependencies(Environments.getArtifacts(options))
```

Comment:
Would this be better placed in createOrGetDefaultEnvironment?

Reply:
agreed. moved to createOrGetDefaultEnvironment.